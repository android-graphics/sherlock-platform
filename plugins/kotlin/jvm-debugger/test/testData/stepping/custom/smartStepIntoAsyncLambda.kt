package smartStepIntoAsyncLambda

fun lambdaInSequence() {
    val ranges = listOf(1, 2, 3)
    // SMART_STEP_INTO_BY_INDEX: 1
    // STEP_OUT: 1
    // RESUME: 1
    //Breakpoint!
    ranges.asSequence()
        .map {
            it
        }
        .forEach(::println)
}

fun lambdaInStream() {
    val ranges = listOf(1, 2, 3)
    // SMART_STEP_INTO_BY_INDEX: 1
    // STEP_OUT: 1
    //Breakpoint!
    ranges.stream()
        .map {
            it
        }
        .forEach(::println)
}

fun main() {
    lambdaInSequence()
    lambdaInStream()
}
