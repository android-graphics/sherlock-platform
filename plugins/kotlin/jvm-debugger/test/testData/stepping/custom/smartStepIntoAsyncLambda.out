LineBreakpoint created at smartStepIntoAsyncLambda.kt:9
LineBreakpoint created at smartStepIntoAsyncLambda.kt:21
Run Java
Connected to the target VM
smartStepIntoAsyncLambda.kt:9
smartStepIntoAsyncLambda.kt:11
smartStepIntoAsyncLambda.kt:13
smartStepIntoAsyncLambda.kt:21
smartStepIntoAsyncLambda.kt:23
smartStepIntoAsyncLambda.kt:26
Disconnected from the target VM

Process finished with exit code 0
